src/formatting.rs

```diff
@@ -147,19 +156,34 @@ fn insert_newline(mut bytes: Vec<u8>) -> Vec<u8> {
     bytes
 }
 
+/// Given a slice of bytes, strip all spaces, new lines and tabs found within
+///
+/// Because you don't know how the incoming code is formatted,
+/// the best way to format is to strip all the whitespace
+/// and afterwards include the new lines and indentation correctly
+/// according to the configuration
 pub fn trim_ascii_whitespace(x: &[u8]) -> &[u8] {
-    let from = match x.iter().position(|x| !x.is_ascii_whitespace()) {
-        Some(i) => i,
-        None => return &x[0..0],
-    };
+    let Some(from) = x.iter().position(|x| !x.is_ascii_whitespace()) else { return &x[0..0] };
     let to = x.iter().rposition(|x| !x.is_ascii_whitespace()).unwrap();
     &x[from..=to]
 }
 
+/// Returns true if the Block has at least 1 Pipeline
+///
+/// This function exists because sometimes is passed to `nufmt` an empty String,
+/// or a nu code which the parser can't identify something runnable
+/// (like a list of comments)
+///
+/// We don't want to return a blank file if that is the case,
+/// so this check gives the opportunity to `nufmt`
+/// to know when not to touch the file at all in the implementation.
 fn block_has_pipilnes(block: &Block) -> bool {
     !block.pipelines.is_empty()
 }
 
+/// Returns true if the `Span` is the last Span in the slice of `flat`
+///
+/// Takes a Span and a slice of tuples `(Span, FlatShape)`.
```

Comment:
i think this kind of comment is useless because it's the same as the signature of the Rust function :+1: